persistence/sql/common.go

```diff
@@ -60,3 +63,100 @@ func createGetTimerTaskResponse(
 	}
 	return resp, nil
 }
+
+func (p sqlProcessStoreImpl) handleStateDecision(ctx context.Context, tx extensions.SQLTransaction,
+	request data_models.HandleStateDecisionRequest) (*data_models.HandleStateDecisionResponse, error) {
+	hasNewImmediateTask := false
+
+	// these fields will be updated and returned back in response for ProcessExecutionRowForUpdate
+	sequenceMaps := request.ProcessExecutionRowStateExecutionSequenceMaps
+	procExecWaitToComplete := request.ProcessExecutionRowWaitToComplete
+	procExecStatus := request.ProcessExecutionRowStatus
+
+	if len(request.StateDecision.GetNextStates()) > 0 {
+		hasNewImmediateTask = true
+
+		for _, next := range request.StateDecision.GetNextStates() {
+			stateIdSeq := sequenceMaps.StartNewStateExecution(next.StateId)
+
+			stateInputBytes, err := data_models.FromEncodedObjectIntoBytes(next.StateInput)
+			if err != nil {
+				return nil, err
+			}
+
+			stateInfo := data_models.AsyncStateExecutionInfoJson{
+				Namespace:             request.Namespace,
+				ProcessId:             request.ProcessId,
+				ProcessType:           request.ProcessType,
+				WorkerURL:             request.WorkerUrl,
+				StateConfig:           next.StateConfig,
+				GlobalAttributeConfig: request.GlobalAttributeTableConfig,
+			}
+
+			stateInfoBytes, err := stateInfo.ToBytes()
+			if err != nil {
+				return nil, err
+			}
+
+			err = insertAsyncStateExecution(ctx, tx, request.ProcessExecutionId, next.StateId, stateIdSeq, next.StateConfig, stateInputBytes, stateInfoBytes)
+			if err != nil {
+				return nil, err
+			}
+
+			err = insertImmediateTask(ctx, tx, request.ProcessExecutionId, next.StateId, stateIdSeq, next.StateConfig, request.TaskShardId)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// If the process was previously configured to gracefully complete and there are no states running,
+	// then gracefully complete the process regardless of the thread close type set in this state.
+	// Otherwise, handle the thread close type set in this state.
+
+	toGracefullyComplete := procExecWaitToComplete && len(sequenceMaps.PendingExecutionMap) == 0
```

Comment:
just naming nit, maybe:

```
shouldGracefulComplete := procExecGracefulCompleteRequested && len(sequenceMaps.PendingExecutionMap) == 0
```
related to the issue: https://github.com/xcherryio/xcherry/issues/66

`GracefulCompleteRequested` should make it more clear that there was a gracefulComplete decision was made before.

Reply:
I think it will be better to make another PR for the issue, rather than partially renaming something in this PR to make naming inconsistent.